### ISSUE TITLE
Feature Suggestion: Title Case

### ISSUE THREAD

**Issue by alanqthomas** [panzerdp/voca#21]
## Feature Suggestion

Convert a string to title case.

### Examples
```
v.titleCase('hello world')
// => 'Hello World
v.titleCase('Hello world')
// => 'Hello World
v.titleCase('hello World')
// => 'Hello World
```

### Implentation Suggestions
[This SO question has many suggestions.](http://stackoverflow.com/questions/196972/convert-string-to-title-case-with-javascript)

The top answer uses string.replace() with regex: 
```
function toTitleCase(str) {
  return str.replace(/\w\S*/g, function(txt){return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();});
}
```
Not sure about the coverage of all cases.

### Options
In formal title case, words like 'a', 'it', 'of', etc. are ignored (not capitalized). Consider this as an option.
`v.titleCase(subject, allWords)`
```
v.titleCase('this is a title of great proportions', true);
// => 'This Is A Title Of Great Proportions'
v.titleCase('this is a title of great proportions', false);
// => 'This is a Title of Great Proportions'
```

**Comment by panzerdp** [comment]
Hey @alanqthomas,
Interesting idea.

What do you think about the punctuation `. , ?` removal from the sentence? Probably it should be kept in the title case too.

Thanks!

**Comment by alanqthomas** [comment]
@panzerdp I would think all punctuation should be left intact as well.

**Comment by panzerdp** [comment]
Sounds good. I will implement this function.

**Comment by panzerdp** [comment]
Implemented in version `1.2.0`.
See `v.titleCase()` [docs](https://vocajs.com/#titleCase).
